### PR TITLE
device train microbatch size pt 2

### PR DESCRIPTION
**Pull request by mvpatel2000**
# What does this PR do?

- renames `train_device_microbatch_size` to `device_train_microbatch_size` based on feedback
- updates notebook example of AGA
- updates tutorial on AGA

# What issue(s) does this change relate to?

[CO-1466](https://mosaicml.atlassian.net/browse/CO-1466)

### CODE REVIEW

**Review by dakinggg — APPROVED** [review]
LGTM, thanks!

File: docs/source/notes/auto_grad_accum.rst [hunk 1]
```diff
@@ -47,6 +47,8 @@ By setting this single parameter, most OOM errors are a thing of the past, and y
 
 This functionality takes advantage of a unique design within our Trainer -- we ask the dataloader to return the complete batch size (also called the optimization batch size), and we handle slicing that batch into microbatches and feeding it to the model. Since we control the slicing, we can dynamically change the microbatch sizes during training.
 
+As a bonus feature, we also support automatic microbatching for evaluation when using the ``device_eval_microbatch_size='auto'`` setting, so the eval batch size doesn't need to be tuned either when changing GPUs or batch sizes.
```
[the remark this answers is not in the document]
Comment (mvpatel2000) [reply docs/source/notes/auto_grad_accum.rst:50]: typo, should be `device_train_microbatch_size='auto'`

[the remark this answers is not in the document]
Comment (dakinggg) [reply docs/source/notes/auto_grad_accum.rst]: I don't think that is right, this whole paragraph is about eval

[the remark this answers is not in the document]
Comment (dakinggg) [reply docs/source/notes/auto_grad_accum.rst]: oh my bad, disregard last comment.

**Review by nik-mosaic — APPROVED** [review]

**Pull request merged**